arbos/programs/native.go

```diff
@@ -233,11 +231,27 @@ func activateProgramInternal(
 	return info, asmMap, err
 }
 
+// getLocalAsm gets asm for local target and recompiles it if not found.
+// Existence of asms for other configured targets is checked and also recompiled if not found (other targets are needed for multi-target recording).
 func getLocalAsm(statedb vm.StateDB, moduleHash common.Hash, addressForLogging common.Address, code []byte, codehash common.Hash, maxWasmSize uint32, pagelimit uint16, time uint64, debugMode bool, program Program, runCtx *core.MessageRunContext) ([]byte, error) {
```

Comment:
This function is weird now and it's main function doesn't match it's name..
Maybe rename it to something like getCompiledContracts, have it return asmMap, and then take just the local target outside of this function?

Reply:
good point, I renamed it to getCompiledProgram and made it return full asmMap

Reply:
a note here: I had to return `map[rawdb.WasmTarget][]byte{rawdb.LocalTarget(): []byte{}}` from wasm implementation of the `getCompiledProgram` to make checks for local target work in the caller